### ISSUE TITLE
unable to put keys containing unicode chars

### ISSUE THREAD

**Issue by foobargeez** [hexfusion/perl-net-etcd#66]
The underlying `encode_base64()` from `MIME::Base64` expects _bytes_ which is breaking when the key contains unicode chars.  Can we encode key string using `utf-8` along the lines of:

`encode_base64(encode('utf-8', $key));`

Thanks!

**Comment by hexfusion** [comment]
Makes sense to me have you tried it?

**Comment by foobargeez** [comment]
> Makes sense to me have you tried it?

Yeah ... I am doing that currently in my code which works fine.  Let me get a PR ready.

**Comment by hexfusion** [comment]
OK while you are there please add some tests, update change log and maybe add an example to POD. Then I can cut a release for you, thanks!

**Comment by hexfusion** [comment]
@foobargeez how are we doing? If you can give me an idea on timeframe for your PR I can make some time to get this done.
